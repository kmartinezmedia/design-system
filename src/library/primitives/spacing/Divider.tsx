import React, { FC } from 'react';
import { View, ViewProps, StyleSheet } from 'react-native';
import { useTheme } from '@designSystem/theme';

export const Divider: FC<ViewProps> = (props) => {
  const { colors } = useTheme();
  return (
    <View
      style={{ ...styles.divider, backgroundColor: colors.onBackground.line }}
      {...props}
    />
  );
};

const styles = StyleSheet.create({
  divider: {
    height: 1,
    width: '100%',
  },
});
