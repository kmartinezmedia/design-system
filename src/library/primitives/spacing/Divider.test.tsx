import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { Divider } from './Divider';

describe('Divider', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<Divider />);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
