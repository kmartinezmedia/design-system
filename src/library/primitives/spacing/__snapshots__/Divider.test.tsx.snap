// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider default 1`] = `
<View
  style={
    Array [
      Object {
        "height": 1,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#F1F3F5",
      },
    ]
  }
/>
`;
